### DISCUSSION TITLE
Getting error in Singleton import

### DISCUSSION THREAD

**Discussion by ashokkumar88** [adonisjs/core#4444] [Help]
### Package version

6.2.2

### Describe the bug

I have created a singleton in the app_provider.ts in the `boot() {}` method like below. 

```
this.app.container.singleton('App/GraphQL', () => graphql_middlewire);
```

Then in a helper file i imported it `import graphql_middlewire from "App/GraphQL"`. But it is giving error `Cannot find package 'App'`

**Comment by Julien-R44** [comment]
You can find a complete example here:

https://docs.adonisjs.com/guides/ioc-container#basic-example

**Comment by ashokkumar88** [comment]
Actually I am still not getting how to create the singleton and import it. Should i create in the `app_provider.ts` `boot` method or create a separate service for it. An example will be very helpful?

**Comment by e-e** [comment]
@ashokkumar88 does something like this work?

```ts
// providers/app_provider.ts

import { ApplicationService } from "@adonisjs/core/types";
import graphql_middlewire from "#services/graphql_middlewire"; // adjust to your actual import

 export default class AppProvider {
   public static needsApplication = true;

   constructor(protected app: ApplicationService) {}

   public register() {
     // Register your own bindings
     this.app.container.singleton('App/GraphQL', () => new graphql_middlewire());
   }

   public async boot() {}

   public async ready() {
     // App is ready
   }

   public async shutdown() {
     // Cleanup, since app is going down
   }
 }


```

```ts
// types/container.ts

declare module "@adonisjs/core/types" {
  import graphql_middlewire from "#services/graphql_middlewire"; // use the proper import here

  interface ContainerBindings {
    "App/GraphQL": GraphQL;
  }
}
```

```ts
// some/other/file.ts

import app from "@adonisjs/core/services/app";

// ...

const graphql = await app.container.make("App/GraphQL");
```

**Comment by ashokkumar88** [comment]
@e-e Thanks for your response.

Actually i created a `GraphqlService` as singleton and import it as `#services/graphql_service`. It is working fine.
